HIP/hip-583.md

```diff
@@ -10,220 +10,452 @@ status: Accepted
+significant complication. If an account without an alias has already interacted with a smart contract, then it is known
+to the smart contract with the "long-zero" `account-num alias` that all accounts have. If that account were to be
+updated with an alias, then which evm address should be used in the EVM? For compatibility, it should use the
+`account-num alias`, at least with contracts it had already interacted with. However, this may run contrary to some
+expectations. While this is an important user flow we want to support, we defer this support to a future HIP. A full
+specification for this capability will likely require the ability for the user to select which alias is their default,
+and to specify on a per-call basis the alias they want to use when interacting with a smart contract.
+
+User stories
+------------------------------------------------------------------------------------------------------------------------
+
+1. As a user with an `ECDSA` derived account from another chain who uses an Ethereum-compatible wallet, I would like to
+   create my Hedera account using that same ECDSA public key and associated Ethereum compatible `evm address alias`.
+2. As a developer, I would like to create a new account with a `key alias` via the `CryptoCreate` transaction.
+3. As a developer, I would like to create a new account with an `evm address alias` via the `CryptoCreate` transaction.
+4. As a developer, I would like to transfer HBAR or tokens to a Hedera account using their `evm address alias`.
+5. As a Hedera user with an `Ethereum-native wallet`, I would like to receive HBAR or tokens in my account by sharing
+   only my `evm address alias`.
+6. As a Hedera user with an `Ethereum-native wallet`, I would like to send HBAR or tokens from my account to another
+   account on Hedera using a valid 20-byte address.
+7. As a Hedera user with a `Hedera-native wallet`,  I would like to transfer HBAR or tokens to another account using
+   only the recipient's `evm address alias`.
+8. As a user with an Ethereum-native wallet, I want to be able to only share my EVM public address to an exchange, and
+   that exchange to create a hollow account for me by transferring funds to the address.
+
+Specification
+------------------------------------------------------------------------------------------------------------------------
+
+This HIP proposes an expansion to the ways in which an account with an `alias` can be created and referenced to ensure
+compatibility with users coming from existing chains with `ECDSA` keys.
+
+It is thus proposed to:
+- Update `CryptoCreate` transactions to support setting an alias using an `evm address alias`
+- Update `CryptoCreate` transactions to support setting an alias using a `key alias`
+- Update `CryptoTransfer` transactions to support hollow-account creation for accounts with an `evm address alias`
```

Comment:
Should we support creation of contracts with an alias? We would need to update the ContractCreateTransaction.

This gives us a special wrinkle in the system. If I do a crypto transfer and create a hollow account, how do I know that should be an account vs. a contract? How do we make sure the contract isn't accidentally created as an account this way?